src/storage-types/src/sinks.rs

```diff
@@ -646,13 +826,37 @@ pub enum KafkaSinkFormat<C: ConnectionAccess = InlinedConnection> {
     Json,
 }
 
-impl KafkaSinkFormat {
+impl<C: ConnectionAccess> KafkaSinkFormat<C> {
     pub fn get_format_name(&self) -> &str {
         match self {
             Self::Avro(_) => "avro",
             Self::Json => "json",
         }
     }
+
+    fn alter_compatible(&self, id: GlobalId, other: &Self) -> Result<(), StorageError> {
+        if self == other {
+            return Ok(());
+        }
+
+        match (self, other) {
+            (Self::Avro(self_avro), Self::Avro(other_avro)) => {
+                self_avro.alter_compatible(id, other_avro)?
+            }
+            (s, o) => {
+                if s != o {
+                    tracing::warn!(
+                        "KafkaSinkFormat incompatible\nself:\n{:#?}\n\nother:{:#?}",
```

Comment:
rust tip: you can put the parameters in the string directly:

```suggestion
                        "KafkaSinkFormat incompatible\nself:\n{s:#?}\n\nother:{o:#?}",
```

Reply:
Yes, I'm familiar with Rust formatting strings––if this is a style you'd like to enforce, maybe there's a lint for it?

Reply:
Ah, no that's fine. I assumed that you'd prefer the shorter version but it's not something I feel remotely strongly about to even consider a lint :D